Jint/Native/Object/ObjectInstance.cs

```diff
@@ -959,10 +959,9 @@ private object ToObject(ObjectTraverseStack stack)
                     break;
 
                 case ObjectClass.Function:
-                    if (this is FunctionInstance function)
+                    if (this is ICallable function)
```

Comment:
There's a PR trying to hide ICallable and possibly trying to remove it at some point altogether if possible. In spec there are function objects (in Jint FunctionInstance) that may be callable (`IsCallable`)

Reply:
Are you suggesting I should check `IsCallable`? As I see it does `this is ICallable` for everything except `Proxy`. 

Also this function is for converting script objects to their CLR counterpart. I don't think anything other than a function should be converted to a CLR function. So, proxies will not be hitting this line anyway.

As for removing `ICallable`, I can't imagine how it is going to work. If we agree on test cases, let's keep this for now and change when `ICallable` removed?

Reply:
Yeah this was just a note. Proxy is a good example that function object might not be callable, even though it has to implement `ICallable` if the target would happen to be callable. My thinking is that we should be able to get away by having `FunctionInstance` (write your own implementation), `ScriptFunctionInstance` (anything from script) and `ClrFunctionInstance` (basically native function). All these have `Call` method from `FunctionInstance`. IConstructor and ICallable are a bit artificial constructs, is all.

The more test coverage the better. Always comes back biting for advanced usage that people like you have when doing interop. All coverage is much appreciated!

Reply:
Ah, ok. It was `FunctionInstance` before but I noticed bound functions did not derive from `FunctionInstance` and `ICallable` seemed like the only thing that covers all of these cases.

Reply:
Ah the BoundFunction... OK that was the pain point we discussed earlier. Well it's a start to make `ICallable` internal at least..